sdk-development-guidelines.md

```diff
@@ -0,0 +1,200 @@
+    * Note: You do not have to export protocols and categories, notwithstanding their visibility.
+* MAY embed private classes in the .m file that uses them, instead of creating a separate .h/.m pair.
+* If a value is only ever used once, it's ok to not define a constant for it as long as the intent is clear (e.g. values passed to `+[UColor colorWithRed:green:blue:alpha:])` or documented with a comment.
+
+### Coding Style
+
+* Code MUST follow a style guide such as:
+    * [Kotlin's official coding conventions](https://kotlinlang.org/docs/reference/coding-conventions.html)
+        * To configure the IntelliJ formatter according to this style guide, please install Kotlin plugin version 1.5.0 or newer, go to Settings | Editor | Code Style | Kotlin, click on "Set from…" link in the upper right corner, and select "Predefined style / Kotlin style guide" from the menu
+    * [Google Java Style](https://google.github.io/styleguide/javaguide.html)
+    * [Swift](https://github.com/raywenderlich/swift-style-guide)
+    * [Objective-C](https://github.com/raywenderlich/objective-c-style-guide)
+* Existing modules SHOULD follow a coding style guide. Always strive for **consistency** within a project's codebase
+* SHOULD use static code analysis, such as [detekt](https://github.com/arturbosch/detekt), clang-format, [SwiftLint](https://github.com/realm/SwiftLint), or oclint to minimise errors and minimise debates and formatting fix-up commits in PRs. It is RECOMMENDED to integrate these tools with the CI workflow as pre-commit git hooks.
+* SHOULD use quality tools available in common config: [Android](https://github.com/rakutentech/android-buildconfig) and [iOS](https://github.com/rakutentech/ios-buildconfig)
+* For Android <img src="images/android.png" alt="drawing" width="15"/>:
+    * MUST use the [Android Support Annotations](https://developer.android.com/studio/write/annotations) for public APIs
+    * Non-Android Java code MUST use the [Jetbrains Annotations](https://www.jetbrains.com/help/idea/annotating-source-code.html) for public APIs
+
+### Library Design
+
+#### Limit impact on consumers
+* SHOULD limit dependencies as much as possible primarily due to risk of version clashes e.g. SHOULD NOT depend on external networking library but also because the dependency may stop being maintained
+* SHOULD NOT have exported dependency on other SDK modules
+* SHOULD NOT have an exported dependency on other SDK modules e.g. if your module needs authentication, do not depend on Authentication SDK. Instead, require your library consumer to pass an access token string to your API instead.
+* SHOULD NOT have unnecessary exported dependencies, in particular
```

Comment:
Is this Android specific? Cos on iOS we can't have non-exported/hidden dependencies.

Reply:
yes, moved it inside Android section